x-pack/plugins/security_solution/scripts/endpoint/common/fleet_server/fleet_server_services.ts

```diff
@@ -247,113 +247,119 @@ const startFleetServerWithDocker = async ({
-            port,
-            serviceToken,
-            policyId,
-            agentVersion,
-            esUrl: esURL.toString(),
+          .catch((error) => {
+            if (!/no such container/i.test(error.message)) {
+              log.verbose(`Attempt to kill currently running fleet-server container with name [${containerName}] was unsuccessful:
+      ${error}`);
+            }
           });
 
-      await execa('docker', ['kill', containerName])
-        .then(() => {
-          log.info(
-            `Killed an existing container with name [${containerName}]. New one will be started.`
-          );
-        })
-        .catch((error) => {
-          if (!/no such container/i.test(error.message)) {
-            log.verbose(`Attempt to kill currently running fleet-server container with name [${containerName}] was unsuccessful:
-      ${error}`);
-          }
-        });
+        log.verbose(`docker arguments:\n${dockerArgs.join(' ')}`);
 
-      log.verbose(`docker arguments:\n${dockerArgs.join(' ')}`);
+        containerId = (await execa('docker', dockerArgs)).stdout;
 
-      containerId = (await execa('docker', dockerArgs)).stdout;
+        log.info(`Fleet server started`);
 
-      log.info(`Fleet server started`);
+        if (!isServerless) {
+          await addFleetServerHostToFleetSettings(kbnClient, log, fleetServerUrl);
+        }
 
-      if (!isServerless) {
-        await addFleetServerHostToFleetSettings(kbnClient, log, fleetServerUrl);
-      }
+        await updateFleetElasticsearchOutputHostNames(kbnClient, log);
 
-      await updateFleetElasticsearchOutputHostNames(kbnClient, log);
+        if (isServerless) {
+          log.info(`Waiting for server [${hostname}] to register with Elasticsearch`);
+          await waitForFleetServerToRegisterWithElasticsearch(kbnClient, hostname, 180000);
+        } else {
+          await waitForHostToEnroll(kbnClient, log, hostname, 120000);
+        }
 
-      if (isServerless) {
-        log.info(`Waiting for server [${hostname}] to register with Elasticsearch`);
+        fleetServerVersionInfo = isServerless
+          ? // `/usr/bin/fleet-server` process does not seem to support a `--version` type of argument
+            'Running latest standalone fleet server'
+          : (
+              await execa('docker', [
+                'exec',
+                containerName,
+                '/bin/bash',
+                '-c',
+                '/usr/share/elastic-agent/elastic-agent version',
+              ]).catch((err) => {
+                log.verbose(
+                  `Failed to retrieve agent version information from running instance.`,
+                  err
+                );
+                return { stdout: 'Unable to retrieve version information' };
+              })
+            ).stdout;
+      } catch (error) {
+        if (retryAttempt < 1) {
+          retryAttempt++;
+          log.info(`Failed to start fleet server, retrying. Error: ${error.message}`);
```

Comment:
this probably should be `log.error()`

also, can you add a second log entry and dump the entire error:

```
log.verbose(dump(error));
```

Reply:
Changed